lib/sensors/stripe_sensor.hpp

```diff
@@ -0,0 +1,20 @@
+#include <optional>
+
+#include <io/hardware_gpio.hpp>
+
+#include <core/logger.hpp>
+
+namespace hyped::sensors {
+    class Sensor{
+        public:
+            Sensor(const std::uint8_t newPin);
+            ~Sensor();
+
+            int getStripeCount();
+            void updateStripes();
+        private:
+            std::uint8_t pin;
+            int stripeCount = 0;

```

Comment:
Should this not be initialized in constructor?

Reply:
Also given that the stripe count starts at zero and is strictly increasing, either `uint16_t` or `uint32_t` would probably be more appropriate?